Android build and nuget

CedricGuillemet opened the pull request:
- Android build script (needs NDK and Ninja)
- Nuget script
- Fixed unitest wstring build
- not using uuid library as there is a simple way to get a uuid on Android

CedricGuillemet committed:
Android build without libuuid dependency
Tests disabled for Android

CedricGuillemet committed:
nuget scripts for Android

CedricGuillemet committed:
wrong nupkg name fix

CedricGuillemet committed:
enable tests for Android
fixed wstring cast

kokuda commented on androidbuild.ps1:
```diff
@@ -0,0 +1,138 @@
+<#
```
This file should go into the cmake folder and use a similar naming convention to the files already there.  We might **one day** want to push these back to the source repo and they should conform to the original naming.

CedricGuillemet replied:
Build script moved

kokuda commented on nuget/android/Lib3MF.android.nuspec.in:
```diff
@@ -0,0 +1,37 @@
+    <!-- Build (C++) -->
+    <file src="${CMAKE_CURRENT_SOURCE_DIR}/include/**/*" target="/build/native/include" />
+    <!-- CMake Module -->
+    <file src="${CMAKE_CURRENT_SOURCE_DIR}/nuget/android/FindLib3MF.android.cmake" target="" />
```
Consider using "FindLib3MF.Android.cmake" instead of "android" in the name. We currently have "Win32", "macOS", "iOS", so "Android" fits better with that naming convention.

CedricGuillemet replied:
Done

kokuda commented on UnitTests/Source/UnitTest_Extensions.cpp:
```diff
@@ -77,25 +77,25 @@ namespace NMR {
 		EXPECT_TRUE(bIsSupported) << L"Materials-Spec not supported";
 		errString = std::wstring(L"invalid 3MF API interface version for Materials-Spec: " + std::to_wstring(NMR_APIVERSION_INTERFACE_MATERIALSPEC));
-		EXPECT_EQ(nAPIInterfaceVersion, (DWORD)(NMR_APIVERSION_INTERFACE_MATERIALSPEC)) << errString;
+		EXPECT_EQ(nAPIInterfaceVersion, (DWORD)(NMR_APIVERSION_INTERFACE_MATERIALSPEC)) << errString.c_str();
```
Curious what the problem was here?  Does getting the w_char* from wstring really fix the issue?

CedricGuillemet replied:
yes, getting wchar* solves the issue...or enabling use of std::wstring (-DGTEST_HAS_STD_WSTRING) as I did

CedricGuillemet committed:
- Moved and rename android build script
- renamed android/FindLib3MF.Android.cmake
- preprocessor definition for wstring overloading
